main.py

```diff
@@ -163,9 +165,26 @@ def handle_errors(self, event):
         The event is sent only if the error is related to a request made
         by flow_manager.
         """
-        xid = event.content["message"].header.xid.value
-        error_type = event.content["message"].error_type
-        error_code = event.content["message"].code
+        message = event.content["message"]
+
+        connection = event.source
+        switch = connection.switch
+
+        xid = message.header.xid.value
+        error_type = message.error_type
+        error_code = message.code
+        error_data = message.data.pack()
+
+        # Get the packet responsible for the error
+        error_packet = connection.protocol.unpack(error_data)
+
+        if message.code == BadActionCode.OFPBAC_BAD_OUT_PORT:
+            for action in error_packet.actions:
```

Comment:
This only works on OF 1.0. In IF 1.3 the flow mod does not have actions, it has instructions, and the instructions may have actions.